CPFCNPJTools/Source/CPFManager.swift

```diff
@@ -0,0 +1,194 @@
+    /// let cpf = cpfManager.generate()
+    /// print(cpf)  // Randomly generated CPF
+    /// ```
+    private func generate() -> String {
+        var get9RandomNumbers = (0..<9).compactMap({ _ in Int.random(in: 0...9) })
+        
+        let calculated1stCheckDigit = Double(calculateCPFCheckSum(cpfBaseDigits: get9RandomNumbers, multiplyBy: 10)).truncatingRemainder(dividingBy: 11)
+        calculated1stCheckDigit < 2 ? get9RandomNumbers.append(0) : get9RandomNumbers.append(11 - Int(calculated1stCheckDigit))
+        
+        let calculated2ndCheckDigit = Double(calculateCPFCheckSum(cpfBaseDigits: get9RandomNumbers, multiplyBy: 11)).truncatingRemainder(dividingBy: 11)
+        calculated2ndCheckDigit < 2 ? get9RandomNumbers.append(0) : get9RandomNumbers.append(11 - Int(calculated2ndCheckDigit))
+        
+        let generatedFakeCPF = get9RandomNumbers.map({ String($0) }).joined()
+        
+        return validate(cpf: generatedFakeCPF) == .valid ? generatedFakeCPF : generate()
+    }
+    
+    /// Generates a random valid CPF and returns it already formatted with the mask (XXX.XXX.XXX-XX).
+    ///
+    /// - Returns: The generated CPF formatted with the mask.
+    ///
+    /// **Example:**
+    /// ```swift
+    /// let cpfManager = CPFManager()
+    /// let formattedCpf = cpfManager.generateMasked()
+    /// print(formattedCpf)  // Example output: "111.444.777-35"
+    /// ```
+    public func generateMasked() -> String? {
+        return format(generate())
+    }
+    
+    /// Formats a CPF in the traditional format `XXX.XXX.XXX-XX`.
+    ///
+    /// - Parameter cpf: The CPF to be formatted. The CPF must have exactly 11 numeric digits.
+    /// - Returns: The formatted CPF as a string or `nil` if the CPF does not have 11 digits.
+    ///
+    /// **Example:**
+    /// ```swift
+    /// let cpfManager = CPFManager()
+    /// let formattedCpf = cpfManager.format("11144477735")
+    /// print(formattedCpf)  // "111.444.777-35"
+    /// ```
+    private func format(_ cpf: String) -> String? {
+        guard cpf.count == 11 else { return nil }
+        let formattedCPF = "\(cpf.prefix(3)).\(cpf.dropFirst(3).prefix(3)).\(cpf.dropFirst(6).prefix(3))-\(cpf.suffix(2))"
+        return formattedCPF
+    }
+    
+    /// Calculates the checksum of the CPF base digits for verifying the CPF digits.
+    ///
+    /// - Parameter cpfBaseDigits: The first 9 digits of the CPF.
+    /// - Parameter multiplyBy: The starting multiplier value (10 for the first digit, 11 for the second).
+    /// - Returns: The sum of the products of the digits and their respective multipliers.
+    ///
+    /// **Example:**
+    /// ```swift
+    /// let cpfManager = CPFManager()
+    /// let checksum = cpfManager.calculateCPFCheckSum(cpfBaseDigits: [1, 2, 3, 4, 5, 6, 7, 8, 9], multiplyBy: 10)
+    /// print(checksum)  // Result of calculation
+    /// ```
+    private func calculateCPFCheckSum(cpfBaseDigits: [Int], multiplyBy: Int) -> Int {
```

Comment:
A private methods doesn't need an example, but that is ok if it has too

Reply:
Done!